### ISSUE TITLE
Display a placeholder(preview) before buffering

### ISSUE THREAD

**Issue by maiconhellmann** [google/ExoPlayer#8761]
I need to show a video placeholder without start buffering when using mobile data network.
I came up with a workaround that sets the buffer size of LoadControl to 1000ms and some visual components to set it back to its default value afterwards.

Is there a better way to achieve this?

Last releases the API has changed a bit with lots of deprecations.
In my opinion, the documentation is not very clear on how to use all those factories to achieve specific results although individually the java docs are quite detailed.

**Comment by lcf87** [comment]
Please take a look at #2874. What you did makes sense to me.

As for the doc, I agree we lack some clarity on the higher level ones, may I ask which one is it that you find hard to follow?

**Comment by ojw28** [comment]
If you control the serving side, then this would normally be done by extracting thumbnails on the serving side (e.g., when the video is uploaded or ingested) and then serving them as regular images (rather than videos).

If you don't control the serving side, then I think https://github.com/google/ExoPlayer/issues/3609 is tracking fixing this properly. Depending on what kind of media you're playing, you may also find https://developer.android.com/reference/android/media/MediaMetadataRetriever useful.

Closing as a duplicate of https://github.com/google/ExoPlayer/issues/3609.
